Bug fix: `ibuffer-projectile-root' did not return nil when the file was not in a project.

bcoconni opened the pull request:
It returned `default-directory' instead which result in ibuffer pollution with an non existing project [~/].

bcoconni committed:
Bug fix: `ibuffer-projectile-root' did not return nil when the file was not in a project.

bcoconni committed:
Bug fix #2: Discard files that do not have a projectile root. If `projectile-require-project-root' is set to nil then `projectile-project-root' return the default-directory instead of nil for files not in a project.
Use `ignore-errors' instead.

bcoconni committed:
Projectile groups are named according to the project name instead of the projectile root path.

purcell commented on ibuffer-projectile.el:
```diff
@@ -98,10 +98,16 @@ If the file is not in a project, then nil is returned instead."
+                (delq nil (mapcar (lambda (buf)
+                                    (let ((root (ibuffer-projectile-root buf)))
+                                      (when root
+                                        (cons (with-current-buffer buf
```
Rather than splice extra projectile information into the items here, please simply modify `ibuffer-projectile-root` to return a cons cell for each entry. (`ibuffer-vc` uses the same approach.)

bcoconni replied:
Oh I did not realize that each new commit to that branch in my fork would pile up in the pull request.

Anyway, I pushed a new commit 6a0530b following your comment and tried to mimic `ibuffer-vc-root`. Hopefully, it should address your concern.

If you want me to rebase these commits for a cleaner revision history, let me know.

purcell reviewed (changes requested):
Thanks for this!

bcoconni committed:
Simplified the code as per Steve Purcell suggestion.

Also `ibuffer-projectile--include-file-p' is called after we have checked that the buffer is backed by a file and that that file is in a project. Otherwise it is not worth the trouble to check if the file should be ignored or not.

purcell commented:
That's great, thanks!
